lib/index.js

```diff
@@ -308,10 +330,19 @@ NodeHttpXHR.prototype = Object.create(
      * This will be incomplete until the response actually finishes.
      *
      * @type {?String}
+     * @throws when `response` not a String
      * @readonly
      */
     responseText: {
-      get: function getResponseText() { return this._responseText; }
+      get: function getResponseText() {
+
+        // @see https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseText#Exceptions
+        if (this._responseType !== 'text' && this._responseType !== '') {
```

Comment:
we could write a private method to check this, since it is done below as well

Reply:
Seems like a good idea 👍